### PR TITLE
Oi 1032  .dispose should expect mandatory status parameter

### PR DESCRIPTION
**Pull request by savkaoleg**

### CODE REVIEW

File: src/ox_modules/module-mob.js [hunk 1]
```diff
@@ -276,6 +276,11 @@ export default class MobileModule extends WebDriverModule {
      * @summary Ends the current session.
      */
     async dispose(status) {
+
+        if (!status) {
+            throw new Error('Status field is required and should be one of passed, failed or canceled');
```
Comment (romovs) [inline src/ox_modules/module-mob.js:281]: Should throw OxygenError with SCRIPT_ERROR instead of throwing generic Error, similiarly how we do for all other situations with invalid arguments.

Also documentation for this method is not updated.

And we should probably accept only `passed` and `failed`. `cancelled` is in internal status which is triggered only when stopping currently running and thus shouldn't be settable by user?

File: src/ox_modules/module-web.js [hunk 2]
```diff
@@ -286,7 +289,8 @@ export default class WebModule extends WebDriverModule {
                         const body = "{\"passed\":"+passed+"}";
 
                         const myAccount = new SauceLabs({ user: username, key: accessKey});
-                        myAccount.updateJob(username, id, body);
+                        await myAccount.updateJob(username, id, body);
```
Comment (romovs) [inline src/ox_modules/module-web.js:292]: This doesn't seem to be related to the current commit?

File: src/ox_modules/module-web.js [hunk 3]
```diff
@@ -271,11 +271,14 @@ export default class WebModule extends WebDriverModule {
      */
     async dispose(status) {
         this._whenWebModuleDispose = defer();
+
+        if (!status) {
```
Comment (romovs) [inline src/ox_modules/module-web.js:275]: See mob remarks.

**Review by romovs — CHANGES_REQUESTED** [review]

File: src/ox_modules/module-mob.js (hunk 1 above)
Comment (savkaoleg) [reply src/ox_modules/module-mob.js]: Fixed

File: src/ox_modules/module-web.js (hunk 2 above)
Comment (savkaoleg) [reply src/ox_modules/module-web.js]: Fixed

File: src/ox_modules/module-web.js (hunk 3 above)
Comment (savkaoleg) [reply src/ox_modules/module-web.js:276]: Fixed
